Src/BootCamp.Chapter/MostCommonLetterFinder.cs

```diff
@@ -1,10 +1,89 @@
-﻿namespace BootCamp.Chapter
+﻿using System;
+using System.Collections.Generic;
+
+namespace BootCamp.Chapter
 {
     public static class MostCommonLetterFinder
     {
         public static char Find(string sentence)
         {
-            return 'z';
+            if (string.IsNullOrWhiteSpace(sentence))
+            {
+                throw new ArgumentNullException();
+            }
+
+            List<char> sentenceList = new List<char>();
+            foreach (char character in sentence)
+            {
+                sentenceList.Add(character);
+            }
+            sentenceList.Sort();
+
+            return GetMostCommonLetter(sentence, sentenceList);
+        }
+
+        private static char GetMostCommonLetter(string originalSentence, List<char> sortedSentenceList)
+        {
+            int mostCommonLetterCount = 0;
+            char mostCommonLetter = ' ';
+            int nextLetterCount = 0;
+            char nextLetter = ' ';
+
+            foreach (char c in sortedSentenceList)
```

Comment:
Interesting solution. However, you can make it much more simple. Have you considered using a dictionary? :)